### PR TITLE
WIP: driftphp/server#28

### PR DESCRIPTION
**Pull request by Deltachaos**
An example how a implementation for streams could look like

### CODE REVIEW

**Comment by mmoreram** [comment]
Hum!

Very very very interesting.
Let me check this PR.
Ignore this php73 error. Check that the error is not due to a PHP version, but sometimes fails.
By rerunning them should be enough :)

File: src/RequestHandler.php [hunk 1]
```diff
@@ -245,12 +261,26 @@ private function toServerResponse(
             }
         }
 
+        if ($symfonyResponse instanceof BinaryFileResponse) {
```
Comment (mmoreram) [inline src/RequestHandler.php]: This is inconsistent, right? I mean, if that response can be another thing but a Symfony Response, then is not `$symfonyResponse` but `$response`.

Comment (Deltachaos) [reply src/RequestHandler.php]: `BinaryFileResponse` extends the Symfony response in the same way as `StreamedResponse` does. 

See https://github.com/symfony/http-foundation/blob/master/StreamedResponse.php and https://github.com/symfony/http-foundation/blob/master/BinaryFileResponse.php

So both objects pass the method type hint but need different handling within the method to benefit from the streaming capabilities.

**Comment by Deltachaos** [comment]
Just to mention again, this is a Work in Progress. There are no testes yet and I just have written some code how i think this could work. I have made a few first tests with a streamed response yesterday and something blocks forever. This is something I still need to fix.
Streamed requests are not tested at all (but are probably a lot easier as the Symfony request object accepts ressources as body parameter). Maybe it is worth to split the PR into one for streamed requests and one for streamed response.

**Comment by mmoreram** [comment]
@Deltachaos great!

Please, if you need any help, don't hesitate to ask me or to join the DriftPHP community and ask there - https://gitter.im/driftphp/community

Thanks for your time and work :)

File: src/RequestHandler.php (hunk 1 above)
Comment (mmoreram) [reply src/RequestHandler.php]: You're right ^^

**Comment by mmoreram** [comment]
@Deltachaos any updates on that? Thanks!

**Comment by mmoreram** [comment]
@Deltachaos I merged this PR - https://github.com/driftphp/server/commit/3ed8edb751f65a2c836a667c3ff82a8ca43f6e32

This commit covers
- Regular Response object
- PSR7 Response object, including Stream ones
- All implementations are currently supporting encoding, even stream ones.

Only Symfony StreamedResponse is missing. Not a must but can be covered in the future. In the meanwhile, we can work properly with the PSR7 one.

I close this PR cue to inactivity.
Thanks :)

**Comment by Deltachaos** [comment]
@mmoreram nice! when i find some time i will add Symfony StreamedResponse

**Pull request closed without merging**